src/core/tests/model.cpp

```diff
@@ -2297,3 +2297,21 @@ TEST(model, create_model) {
     EXPECT_THROW(ov::Model(ov::ResultVector{}, {}, {}, {nullptr}, ""), ov::Exception);
     EXPECT_THROW(ov::Model(ov::OutputVector{ov::Output<ov::Node>{nullptr, 0}}, {}, {}, {}, ""), ov::Exception);
 }
+TEST(model, batch_size_zero) {
+    OV_EXPECT_THROW_HAS_SUBSTRING(
```

Comment:
Thanks for contribution.

It looks like current solution will provide batch `0` inforamtion correctly for static shapes only but there are some cases with dynamic shapes which will be not detected e.g.
```cpp
#include "common_test_utils/file_utils.hpp"
#include "openvino/runtime/core.hpp"
#include "openvino/util/file_util.hpp"

TEST(model, batch_size_zero) {
    auto model = bs_utils::create_n_inputs(ov::element::f32, {ov::PartialShape{-1, -1, -1, -1}}, {"NCHW"});
    auto core = ov::Core{};

    auto plugin_path = ov::util::make_plugin_library_name(ov::test::utils::getExecutableDirectory(),
                                                          std::string("openvino_template_plugin") + OV_BUILD_POSTFIX);

    core.register_plugin(plugin_path, "TEMPLATE");
    auto compiled_model = core.compile_model(model, "TEMPLATE"); // do check for CPU, GPU, others
    auto infer_request = compiled_model.create_infer_request();

    auto input_tensor = ov::Tensor(ov::element::f32, {0, 299, 299, 3});
    infer_request.set_input_tensor(input_tensor);

    infer_request.start_async();
    infer_request.wait();

    ASSERT_TRUE(true);
}
```
For `TEMPLATE` (test) plugin it will work but when `CPU` used test just pass without exception.

We need tests for static and dynamic shapes similar to  this example which will be executed for different plugins (shared test(s) this file can be candidate to add it `src/tests/functional/plugin/shared/src/behavior/ov_infer_request/batched_tensors.cpp`).

Reply:
I have added a check at the inference level if the input is dynamic to handle this issue